### DISCUSSION TITLE
[BUG] Connect-PnPOnline not working using a service principal

### DISCUSSION THREAD

**Discussion by jay-dee-20** [pnp/powershell#4648] [General]
### Notice
Many bugs reported are actually related to the PnP Framework which is used behind the scenes. Consider carefully where to report an issue:

1. **Is the issue related to the cmdlet itself, its parameters, the syntax, or do you suspect it is the code of the cmdlet that is causing the issue?** Then please continue reporting the issue in this repo.

I am trying to use: 

_$connection = Connect-PnPOnline -Url $destinationUrl -ClientId $clientId -ClientSecret $clientSecret -Tenant $tenantId_

And get the error:

_Parameter set cannot be resolved using the specified named parameters. One or more parameters issued cannot be used together or an insufficient number of parameters were provided._

The documentation I've seem seems to indicate this is all that is needed, so I'm wondering if this is a bug?

### Reporting an Issue or Missing Feature
Please confirm what it is that your reporting

### Expected behavior 
Please describe what output you expect to see from the PnP PowerShell Cmdlets

### Actual behavior 
Please describe what you see instead. Please provide samples of output or screenshots.

### Steps to reproduce behavior
Please include complete script or code samples in-line or linked from [gists](https://gist.github.com/)

```
$installed = Get-Module -Name PnP.PowerShell -ListAvailable
If (!$installed) { Install-Module PnP.PowerShell }
Import-Module PnP.PowerShell

$destinationUrl = "<SP Site URL>" 
$clientId = "<client ID>"
$clientSecret = "<client secret>"
$tenantId = "<tenant ID>"

$connection = Connect-PnPOnline -Url $destinationUrl -ClientId $clientId -ClientSecret $clientSecret -Tenant $tenantId
```

### What is the version of the Cmdlet module you are running?
(you can retrieve this by executing ```Get-Module -Name "PnP.PowerShell" -ListAvailable```)

2.12.0

### Which operating system/environment are you running PnP PowerShell on?
- [X] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by jackpoz** [comment]
Try without `-Tenant` parameter, that seems to work for me:

![image](https://github.com/user-attachments/assets/17802e67-bd65-4222-8278-ddc7c3a754c3)

**Reply by gautamdsheth** [reply]
Hey @jay-dee-20 , you can check this example.

https://pnp.github.io/powershell/cmdlets/Connect-PnPOnline.html#example-3

Do note, this method of auth is not recommended to use. We suggest you use ClientId + Certificate.
The ClientId + Secret relies on now deprecated SharePoint ACS flow , it will stop working sometime later when MS removes support for it.
